### PR TITLE
Login: Fix Google login popup being blocked in IE

### LINKED ISSUE #15554: Social signup and login: Google popup blocked on IE
When trying to use the "Continue with Google" option on signup under Internet Explorer with default settings, a subtle notice about a blocked popup appears at the bottom of the screen, and nothing else happens. To continue, users have to allow the site to show popups (which triggers a reload), and then click "Continue with Google" again.

I tested this with VMs from [Microsoft](https://developer.microsoft.com/en-us/microsoft-edge/tools/vms/), under Win8.1/IE11 and Win10/Edge

ranimac3 screenshot 2017-06-27 at 14 13 18

This was originally reported for M1, but looks like it was never posted on GH. 

See also https://github.com/Automattic/wp-calypso/issues/15396, which may be related.

### PR DESCRIPTION
**Pull request by pento**
This pull request fixes #15554 and #15396.

#### Testing instructions

1. Run `git checkout fix/15554-google-login-popup-ie` and start your server, or open a [live branch](https://calypso.live/log-in?branch=fix/15554-google-login-popup-ie)
2. Open the [`Log In` page](http://calypso.localhost:3000/log-in) in IE or Windows Phone browser
3. Click/Tap the Google button, and confirm that the Google popup opens

#### Reviews

- [ ] Code
- [ ] Product

### CODE REVIEW

**Comment by scruffian** [comment]
I tested in IE. The popup shows, but after I authenticate I am not logged in.

**Comment by pento** [comment]
@scruffian: I can't reproduce that - I tested in IE11 and Edge - they both displayed the popup and logged me in correctly.

I also tested that the popup opens on Windows Phone, but didn't login, as I'm not sure that Browserstack is the safest place for me to be entering my passwords. :-)

File: client/components/social-buttons/google.js [hunk 1]
```diff
@@ -117,12 +121,10 @@ class GoogleLoginButton extends Component {
...
 		const { responseHandler } = this.props;
 
-		// Handle click async if the library is not loaded yet
-		// the popup might be blocked by the browser in that case
-		// options are documented here:
+		// Options are documented here:
 		// https://developers.google.com/api-client-library/javascript/reference/referencedocs#gapiauth2signinoptions
-		this.initialize()
-			.then( gapi => gapi.auth2.getAuthInstance().signIn( { prompt: 'select_account' } ).then( responseHandler ) )
+		Promise.resolve( window.gapi.auth2.getAuthInstance().signIn( { prompt: 'select_account' } ) )
```
Comment (stephanethomas) [inline client/components/social-buttons/google.js]: Could you explain why we need to use `Promise.resolve()` here?

As far I can see the [`signIn()` method](https://developers.google.com/api-client-library/javascript/reference/referencedocs#googleauthsignin) already returns a promise.

Comment (stephanethomas) [inline client/components/social-buttons/google.js]: FYI, I've tested this pull request without that call and it works fine.

Comment (pento) [inline client/components/social-buttons/google.js]: Huh, weird. It didn't work for me earlier, but that may've been something else I broke.

**Pull request merged**